### ISSUE TITLE
Test: need a test for the flock(logfile) approach to serializing allocation and log append

### ISSUE THREAD

**Issue by jagalactic** [cxl-micron-reskit/famfs#2]
I'm thinking the test should prove that while the lock is held, other processes can't get the lock. What else? We currently rely on this being bulletproof...

**Comment by arramesh42** [comment]
Adding more context, please correct if it's inaccurate:

We need to serialize space allocations and log append(.log file). This is currently done under flock() on log file.
Add a test that proves that when the log file is under flock() anyone else trying to get the lock is blocked.

**Comment by jagalactic** [comment]
That's right. The flock() serializes both space allocation and log append.
